airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -361,17 +386,25 @@ def _handle_error_resolution(
 
         if error_resolution.response_action == ResponseAction.FAIL:
             if response is not None:
-                error_message = f"'{request.method}' request to '{request.url}' failed with status code '{response.status_code}' and error message '{self._error_message_parser.parse_response_error_message(response)}'"
+                filtered_response_message = filter_secrets(
+                    f"Request (body): '{str(request.body)}'. Response (body): '{self._get_response_body(response)}'. Response (headers): '{response.headers}'."
+                )
+                error_message = f"'{request.method}' request to '{request.url}' failed with status code '{response.status_code}' and error message: '{self._error_message_parser.parse_response_error_message(response)}'. {filtered_response_message}"
             else:
                 error_message = (
                     f"'{request.method}' request to '{request.url}' failed with exception: '{exc}'"
                 )
 
-            raise MessageRepresentationAirbyteTracedErrors(
+            exception = MessageRepresentationAirbyteTracedErrors(
                 internal_message=error_message,
                 message=error_resolution.error_message or error_message,
                 failure_type=error_resolution.failure_type,
             )
+            
+            # ensure the exception message is emitted before raised
```

Comment:
I think I'm missing something. I thought the issue was that when printing the exception, we would not see the `error_message` and we would only see the `error_resolution.error_message`. This is at least what is shown in the example in the description of [the issue](https://github.com/airbytehq/airbyte-internal-issues/issues/10680). How is emitting the message here changing anything to this?

Reply:
The `self._logger.error()` to do the same as `exception.emit_message()` basically. Probably i don't understand what you want here.

Reply:
run

```
poetry run pytest unit_tests/sources/streams/http/test_http.py -k 'test_error_codes_http_stream_error_resolution_with_response_secrets_filtered' -s
```

you will see both are emitted:
- error_message
- error_resolution.error_message

Reply:
Yes but the platform does not show trace messages in the logs which makes it very hard for us to debug. I want to see the error_message in the logs in the platform

Reply:
Done.